tests/api/test_metrics_steps.py

```diff
@@ -0,0 +1,407 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+import pytest
+from snowflake.cli.api.metrics import (
+    CLIMetrics,
+    CLIMetricsInvalidUsageError,
+    _CLIMetricsStep,
+)
+
+
+def test_metrics_steps_initialization_empty():
+    # given
+    metrics = CLIMetrics()
+
+    # when
+
+    # then
+    assert metrics.steps == []
+
+
+def test_metrics_steps_time_is_valid():
+    # given
+    metrics = CLIMetrics()
+
+    # when
+    with metrics.track_step("step1"):
+        pass
+
+    # then
+    assert len(metrics.steps) == 1
+    step1 = metrics.steps[0]
+    assert step1[_CLIMetricsStep.NAME_KEY] == "step1"
```

Comment:
please just merge a bunch of assertions into the same test. There is no need to write a single test case per field in the output.

Reply:
407 lines -> 283 😅